cloudify_azure/resources/network/networkinterfacecard.py

```diff
@@ -153,7 +154,7 @@ def start(**_):
         pubip_id = ip_cfg.get(
             'properties', dict()).get(
             'publicIPAddress', dict()).get('id')
-        if isinstance(pubip_id, basestring):
+        if isinstance(pubip_id, text_type):
```

Comment:
Maybe in this case an early return could improve readability?

ditto (`text_type != basestring`)